core/base/batch_instantiation.hpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_BASE_BATCH_INSTANTIATION_HPP_
+#define GKO_PUBLIC_CORE_BASE_BATCH_INSTANTIATION_HPP_
+
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/batch_ell.hpp>
+#include <ginkgo/core/matrix/batch_identity.hpp>
+#include <ginkgo/core/preconditioner/batch_jacobi.hpp>
+
+
+namespace gko {
+namespace batch {
+
+
+// just make the call list more consistent
+#define GKO_CALL(_macro, ...) _macro(__VA_ARGS__)
+
+#define GKO_BATCH_INSTANTIATE_PRECONDITIONER(_next, ...)            \
+    _next(GKO_INDIRECT(__VA_ARGS__), gko::batch::matrix::Identity); \
```

Comment:
I was using it wrongly.
no, I think it is required here.
in MSVC, they consider the __VA_ARGS__ as one arguments by default.
`GKO_CALL(A_loop, B_loop, ...)` -> A_loop("B_loop, ...").
will just be `"B_loop, ..."(A_1)` `"B_loop, ..."(A_2)` -> only the first one get expanded properly, the B_loop, ... is considered as one arguments.  no `__VA_ARGS__` in A_loop.

I need to use `GKO_INDIRECT` to take the first arguments out of `__VA_ARGS__` when it may have the nested call.